pkg/controller/migration/convert/core.go

```diff
@@ -303,11 +303,28 @@ func handleNodeSelectors(c *components, install *operatorv1.Installation) error
 
 	// check typha nodeSelectors
 	if c.typha != nil {
-		if c.typha.Spec.Template.Spec.Affinity != nil {
-			return ErrIncompatibleCluster{
-				err:       "node affinity not supported for typha deployment",
-				component: ComponentTypha,
-				fix:       "remove the affinity",
+		// we can migrate typha affinities provided they are a NodeAffinity for Preferred.
+		if aff := c.typha.Spec.Template.Spec.Affinity; aff != nil {
+			if aff.PodAffinity != nil || aff.PodAntiAffinity != nil {
```

Comment:
What if the pod anti-affinity matches the one that we're going to install ourselves?

Reply:
The pod anti-affinity you're referring to is applied during the namespace migration, which occurs long after this code here is completed. This code here should never be run when a namespace migration is in progress, so we shouldn't ever block our own migration from happening :+1: